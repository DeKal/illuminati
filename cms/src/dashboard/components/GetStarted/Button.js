import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'

const ItemButton = withStyles(theme => ({
  root: {
    fontSize: theme.fontSizes.M,
    backgroundColor: theme.colors.primary,
    marginTop: theme.spaces.gutter,
    padding: `${theme.spaces.xsmall}px ${theme.spaces.medium}px`,
    textTransform: 'none'
  }
}))(Button)

export default ItemButton
