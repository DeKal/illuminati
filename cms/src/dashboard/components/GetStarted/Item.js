import { styled } from '@material-ui/core/styles'

const Item = styled('div')({
  minHeight: '100px'
})

export default Item
