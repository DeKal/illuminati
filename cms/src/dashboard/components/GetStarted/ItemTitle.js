import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

const ItemTitle = withStyles(theme => ({
  root: {
    fontSize: theme.fontSizes.XL,
    fontWeight: 600
  }
}))(Typography)

export default ItemTitle
