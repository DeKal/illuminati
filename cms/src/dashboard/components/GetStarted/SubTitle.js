import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

const SubTitle = withStyles(theme => ({
  root: {
    fontSize: theme.fonts.L,
    color: theme.colors.secondaryText,
    marginBottom: `${theme.spaces.medium}px`
  }
}))(Typography)

export default SubTitle
