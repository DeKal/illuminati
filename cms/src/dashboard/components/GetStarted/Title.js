import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

const Title = withStyles(theme => ({
  root: {
    fontSize: theme.fontSizes.XXL,
    fontWeight: 600,
    padding: '9px 0 4px 0',
    lineHeight: 1.3
  }
}))(Typography)

export default Title
