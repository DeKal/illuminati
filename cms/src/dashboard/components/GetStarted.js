import React from 'react'
import { styled } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import Container from 'core/components/Container'
import Title from './GetStarted/Title'
import SubTitle from './GetStarted/SubTitle'
import Item from './GetStarted/Item'
import ItemTitle from './GetStarted/ItemTitle'
import { Link } from 'react-router-dom'
import { pages, THEME_URL } from 'core/const/pages'

import PostAdd from '@material-ui/icons/PostAdd'
import Visibility from '@material-ui/icons/Visibility'
import LibraryBooksIcon from '@material-ui/icons/LibraryBooks'
import AccountBoxIcon from '@material-ui/icons/AccountBox'
import SettingsIcon from '@material-ui/icons/Settings'
import PhotoLibraryIcon from '@material-ui/icons/PhotoLibrary'

const GetStarted = () => (
  <Container>
    <Title>Welcome to illuminati!</Title>
    <SubTitle>Weâ€™ve assembled some things to get you started:</SubTitle>
    <Grid container spacing={4}>
      <Grid item md={4} xl={3}>
        <Item>
          <ItemTitle>Get Started</ItemTitle>
          <Line>
            <SettingsIcon style={{ marginRight: '5px' }} />
            <BreakLink to={pages.settings.url}>Open Your Settings</BreakLink>
          </Line>
          <Line>
            <PhotoLibraryIcon style={{ marginRight: '5px' }} />
            <BreakLink to={THEME_URL}>Change your theme</BreakLink>
          </Line>
        </Item>
      </Grid>
      <Grid item md={4} xl={3}>
        <Item>
          <ItemTitle>Next steps</ItemTitle>
          <Line>
            <PostAdd style={{ marginRight: '5px' }} />
            <BreakLink to={pages.post.url}>
              Write your first blog post
            </BreakLink>
          </Line>
          <Line>
            <Visibility style={{ marginRight: '5px' }} />
            <a href={'https://phatho-folio.now.sh//blogs'}>View your site</a>
          </Line>
        </Item>
      </Grid>
      <Grid item md={4} lg={3}>
        <Item>
          <ItemTitle>More actions</ItemTitle>
          <Line>
            <LibraryBooksIcon style={{ marginRight: '5px' }} />
            <BreakLink to={pages.posts.url}>Manage Posts</BreakLink>
          </Line>

          <Line>
            <AccountBoxIcon style={{ marginRight: '5px' }} />
            <BreakLink to={pages.profile.url}>Manage Profile</BreakLink>
          </Line>
        </Item>
      </Grid>
    </Grid>
  </Container>
)

export default GetStarted

const Line = styled('div')({
  margin: '8px 0',
  display: 'flex',
  alignItems: 'center'
})

const BreakLink = styled(Link)({
  overflowWrap: 'break-word'
})
