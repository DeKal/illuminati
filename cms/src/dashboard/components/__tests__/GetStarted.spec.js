import * as React from 'react'
import { shallow } from 'enzyme'
import GetStarted from '../GetStarted'

describe('GetStarted', () => {
  it('should render GetStarted with no error', function() {
    const wrap = shallow(<GetStarted />)
    expect(wrap).toMatchSnapshot()
  })
})
