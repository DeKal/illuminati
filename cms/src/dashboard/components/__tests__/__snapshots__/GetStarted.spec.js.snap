// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStarted should render GetStarted with no error 1`] = `
<Styled(div)>
  <WithStyles(WithStyles(ForwardRef(Typography)))>
    Welcome to illuminati!
  </WithStyles(WithStyles(ForwardRef(Typography)))>
  <WithStyles(WithStyles(ForwardRef(Typography)))>
    Weâ€™ve assembled some things to get you started:
  </WithStyles(WithStyles(ForwardRef(Typography)))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={4}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={4}
      xl={3}
    >
      <Styled(div)>
        <WithStyles(WithStyles(ForwardRef(Typography)))>
          Get Started
        </WithStyles(WithStyles(ForwardRef(Typography)))>
        <Styled(div)>
          <SettingsIcon
            style={
              Object {
                "marginRight": "5px",
              }
            }
          />
          <Styled(Link)
            to="/settings"
          >
            Open Your Settings
          </Styled(Link)>
        </Styled(div)>
        <Styled(div)>
          <PhotoLibraryIcon
            style={
              Object {
                "marginRight": "5px",
              }
            }
          />
          <Styled(Link)
            to="/settings/theme"
          >
            Change your theme
          </Styled(Link)>
        </Styled(div)>
      </Styled(div)>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={4}
      xl={3}
    >
      <Styled(div)>
        <WithStyles(WithStyles(ForwardRef(Typography)))>
          Next steps
        </WithStyles(WithStyles(ForwardRef(Typography)))>
        <Styled(div)>
          <PostAddIcon
            style={
              Object {
                "marginRight": "5px",
              }
            }
          />
          <Styled(Link)
            to="/posts"
          >
            Write your first blog post
          </Styled(Link)>
        </Styled(div)>
        <Styled(div)>
          <VisibilityIcon
            style={
              Object {
                "marginRight": "5px",
              }
            }
          />
          <a
            href="https://phatho-folio.now.sh//blogs"
          >
            View your site
          </a>
        </Styled(div)>
      </Styled(div)>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={3}
      md={4}
    >
      <Styled(div)>
        <WithStyles(WithStyles(ForwardRef(Typography)))>
          More actions
        </WithStyles(WithStyles(ForwardRef(Typography)))>
        <Styled(div)>
          <LibraryBooksIcon
            style={
              Object {
                "marginRight": "5px",
              }
            }
          />
          <Styled(Link)
            to="/posts"
          >
            Manage Posts
          </Styled(Link)>
        </Styled(div)>
        <Styled(div)>
          <AccountBoxIcon
            style={
              Object {
                "marginRight": "5px",
              }
            }
          />
          <Styled(Link)
            to="/profile"
          >
            Manage Profile
          </Styled(Link)>
        </Styled(div)>
      </Styled(div)>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Styled(div)>
`;
